pkg/vmprovider/providers/vsphere/utils/vm_utils.go

```diff
@@ -1,7 +1,7 @@
 // Copyright (c) 2022 VMware, Inc. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-package vsphere
+package utils
```

Comment:
This file is moved to a new place, that's why this package name is changed. I'll rename it.

Reply:
I would keep this package as name `utils` for now? I kind of feel like other place or name doesn't make more sense. 😞  This change refactored some of VM update logic, so now there are two packages referencing these utility functions, one from vmprovider_vm (package vsphere) during VM creation, another from session_vm_update (package session) during VM update and package `vsphere` imports package `session`. Since we are going to remove session package, I don't want to add more functions to it. When we finally remove creation/update from session, we can find a better place for these functions.

Reply:
Don't like that. :( We are accumulating techdebt.  I am not sure when the session code will go away, likely it will stay for decent amount of time.
But considering the urgency of this change, it is okay.  Could you please file a JIRA story and link it to fix the tech debt?

Reply:
Fall back to vmprovider_vm_utils.go, this conservation didn't apply any more. Close this.